model/aro_hcp/v1alpha1/cluster_type.model

```diff
@@ -14,6 +14,251 @@ See the License for the specific language governing permissions and
+//   "cloud_provider": {
+//     "kind": "CloudProviderLink",
+//     "id": "123",
+//     "href": "/api/clusters_mgmt/v1/cloud_providers/123"
+//   }
+// }
+// ```
+//
+// When a cluster is created this is optional, and if used it should contain the
+// identifier of the cloud provider to use:
+//
+// ```json
+// {
+//   "cloud_provider": {
+//     "id": "123",
+//   }
+// }
+// ```
+//
+// If not included, then the cluster will be created using the default cloud
+// provider, which is currently Amazon Web Services.
+//
+// The region attribute is mandatory when a cluster is created.
+//
+// The `aws.access_key_id`, `aws.secret_access_key` and `dns.base_domain`
+// attributes are mandatory when creation a cluster with your own Amazon Web
+// Services account.
 class Cluster {
+	// Name of the cluster. This name is assigned by the user when the
+	// cluster is created. This is used to uniquely identify the cluster
+	Name String
+
+	// DomainPrefix of the cluster. This prefix is optionally assigned by the user when the
+	// cluster is created. It will appear in the Cluster's domain when the cluster is provisioned.
+	DomainPrefix String
+
+	// InfraID is used for example to name the VPCs.
+	InfraID String
+
+	// Link to the _flavour_ that was used to create the cluster.
+	link Flavour Flavour
+
+	// Information about the console of the cluster.
+	Console ClusterConsole
+
+	// Flag indicating if the cluster should be created with nodes in
+	// different availability zones or all the nodes in a single one
+	// randomly selected.
+	// For ARO-HCP Clusters, this attribute is unused, and the control plane
+	// is deployed in multiple availability zones when the Azure region where
+	// it is deployed supports multiple availability zones.
+	MultiAZ Boolean
+
+	// Information about the nodes of the cluster.
+	Nodes ClusterNodes
+
+	// Information about the API of the cluster.
+	API ClusterAPI
+
+	// Link to the cloud provider region where the cluster is installed.
+	link Region CloudRegion
+
+	// DNS settings of the cluster.
+	DNS DNS
+
+	// User defined properties for tagging and querying.
+	Properties [String]String
+
+	// Overall state of the cluster.
+	State ClusterState
+
+	// Flag indicating if the cluster is managed (by Red Hat) or
+	// self-managed by the user.
+	Managed Boolean
+
+	// External identifier of the cluster, generated by the installer.
+	ExternalID String
+
+	// Amazon Web Services settings of the cluster.
+	AWS AWS
+
+	// Google cloud platform settings of the cluster.
+	GCP GCP
+
+	// Microsoft Azure settings of the cluster.
+	Azure Azure
+
+	// Network settings of the cluster.
+	Network Network
+
+	// Proxy.
+	Proxy Proxy
+
+	// Link to an optional _ClusterAutoscaler_ that is coupled with the cluster.
+	link Autoscaler ClusterAutoscaler
+
+	// Additional trust bundle.
+	AdditionalTrustBundle String
+
+	// GCP Network.
+	GCPNetwork GCPNetwork
+
+	// Key used for encryption of GCP cluster nodes.
+	GCPEncryptionKey GCPEncryptionKey
+
```

Comment:
I am thinking we can opportunistically remove things like these that no longer make sense for aro-hcp